### ISSUE TITLE
integrate(sin(m*x)*sin(n*x), x, 0, 2*%pi) unconditionally zero

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2687]
Imported from SourceForge on 2024-07-06 18:07:46
Created by **[jbenda](https://sourceforge.net/u/jbenda/)** on 2024-02-22 14:42:44
Original: https://sourceforge.net/p/maxima/bugs/4265

---

Integrating a product of two sine functions sin(mx) sin(nx) over (0,2pi), where m and n are integers, should give 0 for m != n, and pi for m == n. However, Maxima 5.47.0 outputs 0 unconditionally. Interestingly, when I wrap the integrand into "limit(..., m, n)" the program reports the correct number.
~~~
Loading /home/jakub/.maxima/maxima-init.mac
Maxima 5.47.0 https://maxima.sourceforge.io
using Lisp CLISP 2.49.92 (2018-02-18)
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) declare(m, integer, n, integer)$
(%i2) integrate(sin(m*x)*sin(m*x), x, 0, 2*%pi);
(%o2)                                 %pi
(%i3) integrate(sin(m*x)*sin(n*x), x, 0, 2*%pi);
(%o3)                                  0
(%i4) integrate(limit(sin(m*x)*sin(n*x), m, n), x, 0, 2*%pi);
(%o4)                                 %pi
(%i5) bug_report();

Please report bugs to:
    https://sourceforge.net/p/maxima/bugs
To report a bug, you must have a Sourceforge account.
Please include the following information with your bug report:
-------------------------------------------------------------
Maxima-version: "5.47.0"
Maxima build date: "2023-06-06 11:24:35"
Host type: "x86_64-suse-linux-gnu"
Lisp implementation type: "CLISP"
Lisp implementation version: "2.49.92 (2018-02-18) (built on build75 [127.0.0.1])"
User dir: "/home/jakub/.maxima"
Temp dir: "/tmp"
Object dir: "/home/jakub/.maxima/binary/5_47_0/clisp/2_49_92__2018_02_18___built_on_build75__127_0_0_1__"
Frontend: false
-------------------------------------------------------------
The above information is also reported by the function 'build_info()'.
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:07:47
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2024-02-22 20:10:37
Original: https://sourceforge.net/p/maxima/bugs/4265/#0dfa

---

- **labels**:  --> integrate, defint

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:07:50
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2024-03-27 17:00:58
Original: https://sourceforge.net/p/maxima/bugs/4265/#b021

---

I think the  issue comes from `monstertrig` with these lines:
```
        ;; We have a*sin(m*x)*sin(n*x).
        ;; The integral is: a*(sin((m-n)*x)/(2*(m-n))-sin((m+n)*x)/(2*(m+n))
        (return (subliss y
                         '((mtimes) a
                           ((mplus)
                            ((mquotient)
                             ((%sin) ((mtimes) ((mplus) m ((mtimes) -1 n)) x))
                             ((mtimes) 2 ((mplus) m ((mtimes) -1 n))))
                            ((mtimes) -1
                             ((mquotient)
                              ((%sin) ((mtimes) ((mplus) m n) x))
                              ((mtimes) 2 ((mplus) m n)))))))
```
I think we need to ask if `n=m` here.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:07:54
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2024-03-27 17:02:15
Original: https://sourceforge.net/p/maxima/bugs/4265/#147c

---

- **assigned_to**: Raymond Toy

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:07:58
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2024-03-27 17:02:15
Original: https://sourceforge.net/p/maxima/bugs/4265/#a65a

---

Fixed in [0ff2b7e1cb4b4c3c91f9a623d2e014bfd4aaee2b]

We ask the user if n=m before returning a result.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:08:01
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2024-03-27 17:02:36
Original: https://sourceforge.net/p/maxima/bugs/4265/#7aff

---

- **status**: open --> closed
